### ISSUE TITLE
`no-declarations-before-early-exit`: invalid reports with asynchronous code

### ISSUE THREAD

**Issue by pawelblaszczyk5** [sindresorhus/eslint-plugin-unicorn#3373]
### Description

Hello 👋

I've migrated to latest version (awesome work btw) and spotted unexpected issue with `no-declarations-before-early-exit`. It doesn't correctly work with asynchronous code, where value being checked in the early exit can indeed change before and after statement. It breaks a common pattern of using AbortController for guarding asynchronous code aborts.

Thanks in advance!

### Examples

```ts
const test = async (signal: AbortSignal) => {
	const result = await doSomethingAsynchronous(); // 💥 error is reported here

	if (signal.aborted) {
		return;
	}

	await doSomethingElse(result);
};
```

This code is being reported with the error but it's unsafe to move the check before the assignment and it misses the semantics.

**Comment by sindresorhus** [comment]
Make sure you are on v68. This was already fixed there.

**Comment by pawelblaszczyk5** [comment]
Ah, you're right, it works. I forgot about `pnpm` gating me newest version in repository where I checked this. Sorry for missing that 😅
